File: csharp/test/Ice/plugin/PluginInitializeFailFactory.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.plugin;

public class PluginInitializeFailFactory : Ice.PluginFactory
{
    public string pluginName => "Test";

    public Plugin create(Communicator communicator, string name, string[] args) => new PluginInitializeFail();

    internal class PluginInitializeFail : Plugin
    {
        public void initialize() => throw new PluginInitializeFailException();

        public void destroy() => Test.TestHelper.test(false);
    }
}

File: java/test/plugins/src/main/java/test/Ice/plugin/plugins/PluginInitializeFailFactory.java
// Copyright (c) ZeroC, Inc.

package test.Ice.plugin.plugins;

import com.zeroc.Ice.Communicator;
import com.zeroc.Ice.Plugin;
import com.zeroc.Ice.PluginFactory;

public class PluginInitializeFailFactory implements PluginFactory {
    @Override
    public Plugin create(
            Communicator communicator, String name, String[] args) {
        return new PluginInitializeFail();
    }

    static class PluginInitializeFail implements Plugin {
        @Override
        public void initialize() {
            throw new PluginInitializeFailException();
        }

        @Override
        public void destroy() {
            test(false);
        }

        private static void test(boolean b) {
            if (!b) {
                throw new RuntimeException();
            }
        }
    }
}
